src/helpers/lis.rs

```diff
@@ -0,0 +1,98 @@
+//! # Longest Increasing Subsequence
+
+/// # Example
+/// ```
+/// use programming_team_code_rust::helpers::lis::Lis;
+///
+/// let a = [3, 3, 2, 3];
+///
+/// let mut lis = Lis::default();
+/// for &num in &a {
+///    lis.push(num);
+/// }
+///
+/// assert_eq!(lis.dp.len(), 2);
+/// assert_eq!(lis.get_lis(), [2, 3]);
+///
+/// lis.pop();
+///
+/// assert_eq!(lis.dp.len(), 1);
+/// assert_eq!(lis.get_lis(), [2]);
+/// ```
+pub struct Lis<T> {
+    /// dp\[i\].0 = smallest number such that there exists a LIS of length i+1 ending in this number
+    /// dp\[i\].1 = index in original array of dp\[i\].0
+    pub dp: Vec<(T, usize)>,
+    #[allow(clippy::type_complexity)]
+    st: Vec<(Option<usize>, Option<(usize, (T, usize))>)>,
+}
+
+impl<T> Default for Lis<T> {
+    fn default() -> Self {
+        Self {
+            dp: Vec::new(),
+            st: Vec::new(),
+        }
+    }
+}
+
+impl<T: Copy + Ord> Lis<T> {
+    /// append new_elem onto back of vec
```

Comment:
super nit I would rather you say "Pushes `new_elem` onto back of vec`

I like keeping the language of the docs and code tightly coupled